Documentation/MixedRealityConfigurationGuide.md

```diff
@@ -359,23 +353,99 @@ using UnityEngine;
+
+```csharp
+MixedRealityToolkit.Instance.ActiveProfile = ProfileToUse;
+```
+
+Note when setting `ActiveProfile` during runtime, the destroy of the currently running services will happen after the last LateUpdate() of all services, and the instantiation and initialization of the services associated with the new profile will happen before the first Update() of all services.
+
+A noticable application hesitation may occur during this process. Also any script with higher priority than the `MixedRealityToolkit` script can enter its Update before the new profile is properly setup. See [Script Execution Order settings](https://docs.unity3d.com/Manual/class-MonoManager.html) for more information on script priority.
+
+Special handling is required when Unity UI elements are present in the scene. Such elements are usually children of `Canvas` that often requires a reference to the world camera (UI camera). During an active profile switch, UI cameras are destroyed and recreated and thus the references to them have to be updated for all canvases with previously assigned world camera. `VerifyCanvasConfiguration` in the `CanvasUtility` script can be used to update these references. The following script is an example of one way to handle this problem.
+
+
+```csharp
+/// <summary>
+/// Sample MonoBehaviour that will change the active profile and update the references
+/// to UI camera for Canvases to ensure the functionality of Unity UI after the change.
+/// </summary>
+/// <remarks>
+/// Note that this script assumes there is no instantiation and/or destroy of Canvases.
+/// Canvases to be used with MRTK have the CanvasUtility script attached.
+/// </remarks>
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.Input.Utilities;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ActiveProfileSwapper : MonoBehaviour
+{
+    
+    public MixedRealityToolkitConfigurationProfile ProfileToUse;
+    private CanvasUtility[] allCanvases;// All MRTK canvases (i.e. with the CanvasUtility script) in scene
+    private List<CanvasUtility> cameraNotNullCanvases; // Canvases with not-null cameras
+
+    public void ChangeActiveProfile()
+    {
+        if (cameraNotNullCanvases == null)
+        {
+            // Find canvases with not-null world cameras
+            cameraNotNullCanvases = new List<CanvasUtility>();
+            foreach (var canvas in allCanvases)
+            {
+                if (canvas.GetComponent<Canvas>().worldCamera != null)
+                {
+                    cameraNotNullCanvases.Add(canvas);
+                }
+            }
+        }
+        MixedRealityToolkit.Instance.ActiveProfile = ProfileToUse;
+        StartCoroutine(UpdateCanvasCameraReference());
+    }
+
+    private IEnumerator UpdateCanvasCameraReference()
+    {
+        // Wait for the profile change to complete
+        yield return null; // Skip the frame the profile swap started
```

Comment:
Instead of waiting some number of frames explicitly, could this be something like

```c#
while (MixedRealityToolkit.Instance.ActiveProfile != ProfileToUse)
{
    yield return null;
}
```

since it looks like the backing `activeProfile` doesn't get set immediately

Reply:
`activeProfile` is actually set before all the new services are set up so we have to leave the yield returns here for now.